src/tools/w3dviewer/assetinfo.cpp

```diff
@@ -16,14 +16,20 @@
 #include "assetmgr.h"
 
 AssetInfoClass::AssetInfoClass(const char *name, AssetType type, RenderObjClass *robj, TextureClass *texture) :
-    m_name(name), m_type(type), m_texture(texture), m_renderObj(nullptr)
+    m_name(name), m_type(type), m_texture(texture), m_renderObj(robj)
 {
-    Ref_Ptr_Set(m_renderObj, robj);
+    if (m_texture != nullptr)
+        m_texture->Add_Ref();
+
+    if (m_renderObj != nullptr)
+        m_renderObj->Add_Ref();
+

```

Comment:
This is incorrect, original code does not Add Ref or Release on the texture.

Reply:
Fixed